Updated docs for 3.9 web console

spadgett opened the pull request:
The console has been split out into its own deployment running on the
platform in 3.9. Extensions are no longer loaded as files on the master,
but using URLs. Update the docs for the install and customization
changes.

/cc @ahardin-rh @jwforres

spadgett commented:
I plan to add upgrade instructions for extensions in a separate PR.

@jwforres @ahardin-rh This should be ready for review.

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -22,66 +22,81 @@ behavior of the web console and customize it for your needs.
-You can change the look and feel of nearly any aspect of the user interface in
-this way.
+
+[NOTE]
```
s/NOTE/CAUTION

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -22,66 +22,81 @@ behavior of the web console and customize it for your needs.
+As of {product-title} 3.9, extension scripts and stylesheets can be hosted at
+any `https://` URL as long as the URL is accessible from the browser. The files
+might be hosted from a pod on the platform using a publicly accessible route,
+or on another server outside of OpenShift.
```
s/OpenShift/{product-title}

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -22,66 +22,81 @@ behavior of the web console and customize it for your needs.
+To add stylesheets, update the `extensions.stylesheetURLs` property. The value
+is an array of URLs.
+
+Here is an example:
```
s/Here is an example:/ .Example `extensions.stylesheetURLs` Setting

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -22,66 +22,81 @@ behavior of the web console and customize it for your needs.
+    [...]
 ----
 
+After saving the config map, the web console containers will be updated
```
s/config map/ConfigMap

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -22,66 +22,81 @@ behavior of the web console and customize it for your needs.
+`Content-Type: text/css`.
+====
+
+It's a best practice to wrap extension scripts in an Immediately Invoked
```
s/It's/It is

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -147,45 +174,45 @@ angular.module("mylinkextensions", ['openshiftConsole'])
 containing a title and an `href`. These will be turned into links. You can
 completely override the array, push or pop additional links, or modify the
-attributes of existing links.
+attributes of existing links. For example,
```
For example:

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -1183,41 +867,36 @@ many modifications to the web console can be configured using
-- xref:serving-static-files[`openshift_master_extensions`]
+- xref:loading-custom-scripts-and-stylesheets[`openshift_web_console_extension_script_urls`]
+- xref:loading-custom-scripts-and-stylesheets[`openshift_web_console_extension_stylesheet_urls`]
 - xref:serving-static-files[`openshift_master_oauth_template`]
```
The serving-static-files section was removed, so we should remove this link to

spadgett replied:
Yeah, this link was not correct to begin with. It should link to customizing the login page.

ahardin-rh commented on install_config/web_console_customization.adoc:
```diff
@@ -1183,41 +867,36 @@ many modifications to the web console can be configured using
 
-# Configure extensions in the master config for console customization
+# Configure a custom login template in the master config
 # See: https://docs.openshift.com/enterprise/latest/install_config/web_console_customization.html#serving-static-files
```
The serving-static-files section was removed, so we should remove this link to

ahardin-rh commented:
@spadgett Just a few comments from me. Thanks for this!

spadgett commented:
Thanks for the review! I think I addressed your comments.

ahardin-rh commented:
Thanks! Changes LGTM. Can you please squash this into 1 commit? Then, we should be all set.

spadgett commented:
Thanks, @ahardin-rh. Rebased and squashed. I did make one small change to document the `openshift_web_console_install` variable:

https://github.com/openshift/openshift-docs/pull/8009/files#diff-fe94157edddadf88425cf07b556a00a3R1844

ahardin-rh commented:
Looks great! Thanks!

ahardin-rh commented:
/cherrypick enterprise-3.9
